### ISSUE TITLE
Ordering on X axis on the distribution bar chart

### ISSUE THREAD

**Issue by Frituurpanda** [apache/incubator-superset#519]
Hi all,

I have a very simple question that i'm almost embarrassed to ask. I'm not sure how to order on the X axis in Caravel. It is possible to for instance group by certain names and that works fine. Now however, where trying to group by a certain scale. 

You can view the problem here:
![screen shot 2016-05-26 at 11 32 25](https://cloud.githubusercontent.com/assets/5207803/15570349/e5c8e490-2335-11e6-99ae-09dbb682cf6d.png)

We now grouped the results in this test based on the scalelevel, which is a integer value.
Is it possible to order those results from zero to eighteen (or any kind or sort)?

Thanks in advance for any kind of help, you guys have been amazing!

**Comment by verydxz** [comment]
+1 for this and #525, currently I manually create an extremely small 'order by' metric as the 1st metric... but would be nice to be able to specify one

**Comment by Frituurpanda** [comment]
Hi verydxz, 

Thanks for the answer, could you maybe give an example of how you implemented this?
When looking at the screenshot further up, you'll see that the X axis actually has important values in there. The 0 > 18 references to the scalelevel in this domain. The Y axis refers to a simple count.

You said that you create a extremely small order by metric as the first metric. If i'd add a small order by in the series, i'd lose the values that I have in the X axis right?

**Comment by katemarielewis** [comment]
I managed to create the 'order by' metric but had the same issue as @Frituurpanda that I lost the values that I actually wanted to have on the x axis

**Comment by mrbungie** [comment]
We're having the same issue, anyone has any updates on this?

**Comment by Frituurpanda** [comment]
At this point i still don't have a good solution to this. Aside from using a small custom metric in the database. That's not a good solution though since you'll have to add that for every usecase manually..

**Comment by vera-liu** [comment]
https://github.com/airbnb/caravel/pull/1379

**Comment by ondrejkopicka** [comment]
Please, reopen this issue as it is not solved. After #1379, labels are sorted only alphabetically. In case of numeric labels the resulting order is wrong, e.g. 1, 10, 2, 21, 3 instead of 1, 2, 3, 10, 21.

**Comment by Frituurpanda** [comment]
True, have worked around this issue by setting a custom metric in the database to sort based on that. Not really a scaleable solution but it works for my usecase..

**Comment by suuuch** [comment]
I have the same problem.

**Comment by Hackeruncle** [comment]
I also have the same problem.

**Comment by Hackeruncle** [comment]
@Frituurpanda @vera-liu **I also have the same problem.**

**Comment by vtyran** [comment]
Still only able to sort alphabetically, not  by numerical x value.

**Comment by harryprince** [comment]
I got the same problem @vera-liu

**Comment by harryprince** [comment]
I got the same problem @vera-liu

**Comment by vtyr** [comment]
Still an issue, and with the limitation of running multiple queries (creating temp tables and then selecting from them) being an issue, read: it doesn't work - this is an issue.

**Comment by iasmini** [comment]
I could do it selecting the option "Sort Bars"

![image](https://user-images.githubusercontent.com/5900939/35441004-fce0bcc8-0288-11e8-91a5-cdb704947dac.png)

**Comment by CriMenghini** [comment]
@iasmini What's the plot you selected?
What release of superset are you using? In `Distribution - Bar Chart` I can't see any of the fields your screenshot shows! 

Thank you :-)

**Comment by iasmini** [comment]
Version: {"GIT_SHA": "", "version": "0.17.1"}

![image](https://user-images.githubusercontent.com/5900939/36806054-a1e0886a-1c9d-11e8-995b-798b72531cac.png)

**Comment by benkindt** [comment]
For me it still does only sort alphabetically.
> "version": "0.24.0"
For example: it does not work when ordering months given as strings or numbers.

 I think it would be a great improvement to customize the exact order manually or even change the column labels.

**Comment by ariasjose** [comment]
> I could do it selecting the option "Sort Bars"
> 
> ![image](https://user-images.githubusercontent.com/5900939/35441004-fce0bcc8-0288-11e8-91a5-cdb704947dac.png)

This worked for me, thanks!

**Comment by rzeAkbari** [comment]
![Screenshot 2019-07-05 at 12 36 04](https://user-images.githubusercontent.com/46069014/60717239-8f64dd00-9f21-11e9-8557-508404bc0e8a.png)
I don't know if it is still useful or not, but I did it by setting the value of "order_bars" to true, in the "Parameters" of the edit chart part of my "Distribution bar chart" instance.

**Comment by kalimuthu123** [comment]
sorting the chart based on time 00:00 AM ,01:00 AM ,02:00 AM ,03:00AM

 if(fd.order_time){
        	data.forEach((d) => { 
           d.values.sort(function(a,b){return new Date('1970/01/01 ' + a.x) - new Date('1970/01/01 ' + b.x)});
         });

**Comment by michalrudko** [comment]
I am facing same issue now with Superset 1.0.1 and Bar Chart (cannot find Distributed Bar Chart any more.
There is no option to Sort Bars in Bar Chart - how to make the bars sorted by x axis label instead of metric value?

**Comment by bwgroff** [comment]
So I had a hard time figuring this out too. It's confusing but there are two places that "sort" settings exist in the chart editor. First, in the "Data" view at the bottom. That's not what you want! You want to go to "Customize" and that's where you can find the check box for sort bars. Take a look:
![image](https://user-images.githubusercontent.com/5002606/117983675-2ca2b080-b305-11eb-8db5-f28a432c373e.png)

**Comment by muksah123** [comment]
still after checking sort bar checkbox,it is not sorting bar graph in apache superset

**Comment by joaofonsecapromptly** [comment]
Regarding box plot, there is no Sort bars checkbox and when applying sort by, the query still does not work, even with a specific pre made column/metric to sort
